src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -260,5 +330,26 @@ public override BoundNode VisitInterpolatedString(BoundInterpolatedString node)
             }
             return result;
         }
+
+        [Conditional("DEBUG")]
+        private void AssertNoImplicitInterpolatedStringHandlerConversions(ImmutableArray<BoundExpression> arguments, bool allowConversionsWithNoContext = false)
+        {
+            if (allowConversionsWithNoContext)
+            {
+                foreach (var arg in arguments)
+                {
+                    if (arg is BoundConversion { Conversion: { Kind: ConversionKind.InterpolatedStringHandler }, ExplicitCastInCode: false, Operand: BoundInterpolatedString @string })
+                    {
+                        Debug.Assert(((BoundObjectCreationExpression)@string.InterpolationData!.Value.Construction).Arguments.All(
```

Comment:
nit: would it be simpler to check `ArgumentPlaceholders` rather than looking at the arguments of `Construction`? Or maybe both?

Reply:
I'd rather assert against the thing that is actually going to be emitted.